Merge from upstream v14

Alchez opened the pull request:
**Changelogs:**
- https://github.com/frappe/frappe/releases/tag/v14.40.0
  - https://github.com/frappe/frappe/releases/tag/v14.40.1
  - https://github.com/frappe/frappe/releases/tag/v14.40.2
  - https://github.com/frappe/frappe/releases/tag/v14.40.3
- https://github.com/frappe/frappe/releases/tag/v14.41.0
- https://github.com/frappe/frappe/releases/tag/v14.42.0

barredterra committed:
fix: sort options by translated label

(cherry picked from commit 3d13086c629245120f4c6887e4d4a1e5a01624a0)

barredterra committed:
feat: make option sorting configurable

(cherry picked from commit 49ccc0ab6ebc8bc0ee01b2b5842d17583bc37149)

# Conflicts:
#	frappe/core/doctype/docfield/docfield.json

barredterra committed:
feat: enable "Sort Options" for Type field

(cherry picked from commit 6a22be0739c1aa4c2812cc5af442d1213fd5e83e)

barredterra committed:
fix(Data Import): cast selected value to string

(cherry picked from commit 983a9a6ef9ad20604569a14fb3f887b9f34030c0)

shariquerik committed:
fix: always get parsed date value from a user date formatted date value

(cherry picked from commit 4a8602948f9365ec03772274703ae8d75bd0bc3b)

shariquerik committed:
revert: temp fix to load communications

(cherry picked from commit 0e962078feb87695716b1f8f639c454ee62c3429)

shariquerik committed:
fix: add timestamp attr on timeline-item

(cherry picked from commit 82464a41497b6ec7027b6220bef247344b6404f9)

shariquerik committed:
fix: convert string to int

(cherry picked from commit 9e0ed9d2a2bf9c8f09b16429983bc6a6c45519e6)

shariquerik committed:
feat: load more communications button in timeline

(cherry picked from commit b3ab63e697bf9a8df6e63ad11ae29ac9b0706e38)

shariquerik committed:
Merge pull request #21360 from frappe/mergify/bp/version-14-hotfix/pr-21359

ankush committed:
chore: remove automatic transltion portal contribution

barredterra committed:
chore: resolve conflicts

mergify[bot] committed:
perf: index for_user in Notification log (#21379) (#21380)

Missed out before somehow: https://github.com/frappe/frappe/commit/3a5a45d8af6e86742b466e06702d74cc8bade7e3

(cherry picked from commit fcce713b14367ec8c080ea7594df6032137c2920)

Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
Merge pull request #21339 from frappe/mergify/bp/version-14-hotfix/pr-20864

fix: sort options by translated label (backport #20864)

barredterra committed:
fix: sort options in group by field

(cherry picked from commit 3e9a2d01e3e76b416456b2816682631e1bb0e37e)

barredterra committed:
fix: german translations

(cherry picked from commit a2a6eaa15f1da35b69e3025417ce6bb569f27768)

barredterra committed:
fix: make field label translatable

(cherry picked from commit d35099320f389112295162182e4c74bbd38d6ac4)

elifvish committed:
feat: allow no_smtp_authentication to be set from site config (#21344)

mergify[bot] committed:
fix: "Remember Last Selected Value" not working (frappe#21364) (#21395)

(cherry picked from commit b0e8a75ee6de72034db83e224cd381584be6edc9)

Co-authored-by: Marco Fonseca <62901164+treasuryesc@users.noreply.github.com>

mergify[bot] committed:
perf: faster docshare queries (backport #21392) (#21402)

* perf: Index `everyone` in DocShare

A typical share query looks like this.

select `share_name` from `tabDocShare` where `tabDocShare`.`read` = 1.0 and
`tabDocShare`.`share_doctype` = 'Pick List' and (`tabDocShare`.`user` = 'username'
or `tabDocShare`.`everyone` = 1.0) order by `tabDocShare`.`modified` DESC;

None of existing indexes provide `everyone` values quickly, so `OR`
clause effectively clauses full table scan ALL THE TIME.

(cherry picked from commit 85cef317f2bea7ecf81805309d6d0399cdd357bd)

* perf: Dont order_by while fetching shares

Unnecessary confusion to query planner

(cherry picked from commit e9645b759fe32a6aee732b1eb9c7569fefe22422)

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

rtdany10 committed:
fix: missing filter on first connected nagivation

(cherry picked from commit 5cb9795cc2deed22a86727dd24730b38f006d36d)

mergify[bot] committed:
perf: cache document naming rule to avoid recurring db calls (backport #18491) (#21408)

* perf: cache document naming rule to avoid multiple db call

(cherry picked from commit 28a124ca47791d4a42d02668d524c2f83d5b3d37)

* refactor: `get_doctype_map()` using single query and use generator for caching

(cherry picked from commit 78d30905ac34a294a2f0db946c85f05987a47eac)

# Conflicts:
#	frappe/cache_manager.py

* chore: refactor duplicate code

(cherry picked from commit 380a638f3cd8182e857f87497bb73672b4acfaac)

# Conflicts:
#	frappe/cache_manager.py

* fix: dont clear doctype map keys when clearing doctype cache

(cherry picked from commit 85f6f1e01df375692aade92ac55e89b8e9a989b4)

* fix: use property instead of hardcoding

(cherry picked from commit e82e4d1a7344808bbff9a566fda76a6d71f4496a)

* chore: conflicts

---------

Co-authored-by: Daizy <modidaizy5217@gmail.com>
Co-authored-by: Sagar Vora <sagar@resilient.tech>
Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
perf: Faster report exporting logic (#21415)

* perf: Faster report exports to Excel

Try exporting report with 100K rows, most likely it fails or takes a really long time.

Root causes:
- visible_idx was a list, lookups are SLOW AF when lists grow to 100k+
- visible_idx check is not required when I am exporting entire report.

Test with 85,000 rows.

|         | Before | After | Improvement |
| ---     | ---    | ---   | ---         |
| Export  | 25.99  | 0.77  | ~33x faster |

(cherry picked from commit e6d8912ec510002f38c77686f352458f0a30c693)

ankush committed:
Merge pull request #21417 from frappe/mergify/bp/version-14-hotfix/pr-21415

perf: Faster report exporting logic (backport #21415)

blaggacao committed:
fix: replace field replacing correct dom element

(cherry picked from commit 60231c7f358e0f10d19d5bd3ee0cc77eafe14e77)

# Conflicts:
#	frappe/public/js/frappe/form/layout.js

surajshetty3416 committed:
Merge pull request #21423 from frappe/mergify/bp/version-14-hotfix/pr-21305

fix(Data Import): cast selected value to string (backport #21305)

shariquerik committed:
fix: log errors while getting headers and data

(cherry picked from commit 85ac64ddd9794f3f77115b25e51f441041ad0acc)

# Conflicts:
#	frappe/integrations/doctype/webhook/webhook.py

shariquerik committed:
fix: validate webhook secret

(cherry picked from commit 74cc21013fbd954229d98cf8d802ad56706562d3)

mergify[bot] committed:
fix(UX): Indicate that list view is empty because of filters (#21427) (#21428)

(cherry picked from commit 727059a2d115bc37ac3732df0de01ef8e6076a1f)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: make sure number is not zero for `bankers_rounding` (#21431) (#21434)

* fix: make sure num is not zero for bankers_rounding

* test: rounding near zero

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit db6a06d2042fc6e1e512700ca6d9bd6e1f38bf72)

Co-authored-by: Smit Vora <smitvora203@gmail.com>

phot0n committed:
chore: fix linter

phot0n committed:
Merge pull request #21436 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.40.0

# [14.40.0](https://github.com/frappe/frappe/compare/v14.39.0...v14.40.0) (2023-06-20)

### Bug Fixes

* "Remember Last Selected Value" not working (frappe[#21364](https://github.com/frappe/frappe/issues/21364)) ([#21395](https://github.com/frappe/frappe/issues/21395)) ([9874c4a](https://github.com/frappe/frappe/commit/9874c4a07baa9d354e96c2b8cc47eb8b195ffb18))
* add timestamp attr on timeline-item ([3acbe51](https://github.com/frappe/frappe/commit/3acbe51b769b7f36deaedc1ca0dba777d662dca4))
* convert string to int ([494eaf5](https://github.com/frappe/frappe/commit/494eaf575756482f55fc1a89db94799b55e21289))
* **Data Import:** cast selected value to string ([d1343b3](https://github.com/frappe/frappe/commit/d1343b3cb29d5a0e58a94f4374eba69b2ea7282b))
* make sure number is not zero for `bankers_rounding` ([#21431](https://github.com/frappe/frappe/issues/21431)) ([#21434](https://github.com/frappe/frappe/issues/21434)) ([bbf8341](https://github.com/frappe/frappe/commit/bbf8341b9ae4f4174d1030e0bea61122d1673350))
* sort options by translated label ([3e1d9db](https://github.com/frappe/frappe/commit/3e1d9db45f4883216c80bf57e304bedd5e24ad53))
* **UX:** Indicate that list view is empty because of filters ([#21427](https://github.com/frappe/frappe/issues/21427)) ([#21428](https://github.com/frappe/frappe/issues/21428)) ([2b24487](https://github.com/frappe/frappe/commit/2b244875c0bd2a1cdcad2beef2365cff5dfa8f56))

### Features

* allow no_smtp_authentication to be set from site config ([#21344](https://github.com/frappe/frappe/issues/21344)) ([b498114](https://github.com/frappe/frappe/commit/b49811462e1b85124880a80b2c97f1b27b22372e))
* enable "Sort Options" for Type field ([8b07c39](https://github.com/frappe/frappe/commit/8b07c39911556d5a9c3474e059667b77cc848d7e))
* load more communications button in timeline ([d10fc9a](https://github.com/frappe/frappe/commit/d10fc9adab4ee0c5a79d375823df2f0e2b55fd22))
* make option sorting configurable ([57f8076](https://github.com/frappe/frappe/commit/57f8076988537ee9490c7183fbcf8fa5ef059540))

### Performance Improvements

* cache document naming rule to avoid recurring db calls (backport [#18491](https://github.com/frappe/frappe/issues/18491)) ([#21408](https://github.com/frappe/frappe/issues/21408)) ([03417ef](https://github.com/frappe/frappe/commit/03417ef0eb3fc1de6d01cdbe706ea592cffa00b0))
* faster docshare queries (backport [#21392](https://github.com/frappe/frappe/issues/21392)) ([#21402](https://github.com/frappe/frappe/issues/21402)) ([9af99ef](https://github.com/frappe/frappe/commit/9af99efb95319977b3489392d4db5aef92bd466d))
* Faster report exporting logic ([#21415](https://github.com/frappe/frappe/issues/21415)) ([5c2ac47](https://github.com/frappe/frappe/commit/5c2ac47142e6c30ccc2a76d01ac4a8b20d2f7703))
* index for_user in Notification log ([#21379](https://github.com/frappe/frappe/issues/21379)) ([#21380](https://github.com/frappe/frappe/issues/21380)) ([827b829](https://github.com/frappe/frappe/commit/827b829aef251e3f4377d38a34538f16072e9fd3))

### Reverts

* temp fix to load communications ([9ad8b60](https://github.com/frappe/frappe/commit/9ad8b60536ebd1611af7e5a37d47c51d1fc43427))

surajshetty3416 committed:
Merge pull request #21447 from frappe/mergify/bp/version-14-hotfix/pr-21394

fix: sort options in group by field (backport #21394)

mergify[bot] committed:
chore: add translation to label on new doc in Update quick_entry.js (#21444) (#21449)

[skip ci]

(cherry picked from commit 7d352d7c345132d10858480fa9e81036a9f326c3)

Co-authored-by: Ernesto Ruiz <eruiz@wbapps.com>

ernestoruiz89 committed:
chore: Add translation to confirmation text on Update kanban_view.js (#21453)

[skip ci]

(cherry picked from commit 378c97871b7b61ae54aaab4fe77e2e1c7aeef1a4)

ankush committed:
chore: track update actions per doctype

[skip ci]

(cherry picked from commit 4f70200bd5a259095b34e255ed664d58dfdea924)

surajshetty3416 committed:
Merge pull request #21454 from frappe/mergify/bp/version-14-hotfix/pr-21453

Commit:
fix: repeat title value in description of link field.

mergify[bot] committed:
ci: use multiple python version in patch test (backport #21443) (#21457)

* ci: use multiple python version in patch test (#21443)

* ci: bump pyenv action

* ci: set default python version in patch test

* ci: setup pyenv manually

* ci: replace pyenv with setup-action

* debug

* use multiple python versions directly

* Revert "debug"

This reverts commit 692cbc0c9a407b86647ee09079fdde1510f300d7.

* setup py with --python flag

(cherry picked from commit 085eb4124588382ffc4a97f2a78499f3b94d32e3)

# Conflicts:
#	.github/workflows/patch-mariadb-tests.yml

* chore: conflicts

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

ruchamahabal committed:
feat: before/after hooks for any app install/uninstall

(cherry picked from commit ba8f97e7891ce4eff610e3184f94402d5494c4b6)

mergify[bot] committed:
fix: use correct property name for filters in get_report_filters (#21465) (#21468)

(cherry picked from commit 4ad98ecb7f7eb29daafc0af1f1e13a4c70a93fc3)

Co-authored-by: Sayed Ayman <sayedayman203@gmail.com>

ankush committed:
fix(UX): better error message for Encryption key

(cherry picked from commit fb11b5a1b088628a7bebe5bc933be8e26068a73c)

shariquerik committed:
fix: return if exception occur before executing webhook

(cherry picked from commit 3f792a80b1036a7e2c1c28cdd9f1e3f817968f42)

# Conflicts:
#	frappe/integrations/doctype/webhook/webhook.py

shariquerik committed:
fix: removed redundant condition

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>
(cherry picked from commit fe27b8c7e0c622365fe2b034ca5b7bf37bcd7464)

blaggacao committed:
fix: replace_field

splice needs an (integer) index not an element

also section field management must be taken into account for all sections

(cherry picked from commit c415e2781a77c9c030ec29428f690a44580ff888)

# Conflicts:
#	frappe/public/js/frappe/form/section.js

mergify[bot] committed:
chore: Add translation to text in Update kanban_settings.js (#21452) (#21479)

[skip ci]

(cherry picked from commit 84ac0243962fe6bef2d133d1b40d9e214c703341)

Co-authored-by: Ernesto Ruiz <eruiz@wbapps.com>

ankush committed:
chore: add github star CTA on server script sidebar

[skip ci]

(cherry picked from commit ee74830460a13e1042a9f587de423d61b0b3594b)

HarryPaulo committed:
fix: calendar buttons on right get text translated

(cherry picked from commit 9a855c42a5e2edba08ba55207d6761fbd010c255)

mergify[bot] committed:
fix: set prepared report in background (#21485) (#21487)

(cherry picked from commit e2c468df6026992a706116c35074c3bcd6153a2f)

Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
refactor: Deprecate broken-img mixin

(cherry picked from commit 537d5511127235b41b44f7c4fc15df340f89d5de)

surajshetty3416 committed:
Merge pull request #21511 from frappe/mergify/bp/version-14-hotfix/pr-21510

ankush committed:
Merge pull request #21501 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.40.1

## [14.40.1](https://github.com/frappe/frappe/compare/v14.40.0...v14.40.1) (2023-06-28)

### Bug Fixes

* german translations ([3392a9c](https://github.com/frappe/frappe/commit/3392a9c81c1f60630bc3fc6d492403a9356d42f8))
* make field label translatable ([a39dc90](https://github.com/frappe/frappe/commit/a39dc90848346b38391bdb580212d27a2471ffa0))
* set prepared report in background ([#21485](https://github.com/frappe/frappe/issues/21485)) ([#21487](https://github.com/frappe/frappe/issues/21487)) ([c0c9358](https://github.com/frappe/frappe/commit/c0c9358b9664991b6c65b5c5cf98bdb6c73adaf4))
* sort options in group by field ([0ce6777](https://github.com/frappe/frappe/commit/0ce6777d89bbde5a11fc6457cacad1c39bd56d1c))
* use correct property name for filters in get_report_filters ([#21465](https://github.com/frappe/frappe/issues/21465)) ([#21468](https://github.com/frappe/frappe/issues/21468)) ([e6d3855](https://github.com/frappe/frappe/commit/e6d3855c349dd33a52e4513e77accb231d652326))
* **UX:** better error message for Encryption key ([2e2d671](https://github.com/frappe/frappe/commit/2e2d671c74ab2815a19dd151d9f4a9692afa1943))

mergify[bot] committed:
fix(minor): workflow state indicator (#21508) (#21514)

In List view only `Status` type was considered for indicator. Added fields with `Workflow State` as options

(cherry picked from commit 814265d24581ca6024ebdf8c4e53df95ac060726)

Co-authored-by: Maharshi Patel <39730881+maharshivpatel@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'frappe/version-14' into merge-v14

ghost commented:
### Bug Fixes

* sort options by translated label ([3e1d9db](https://github.com/ParsimonyGit/frappe/commit/3e1d9db45f4883216c80bf57e304bedd5e24ad53))
* add timestamp attr on timeline-item ([3acbe51](https://github.com/ParsimonyGit/frappe/commit/3acbe51b769b7f36deaedc1ca0dba777d662dca4))
* convert string to int ([494eaf5](https://github.com/ParsimonyGit/frappe/commit/494eaf575756482f55fc1a89db94799b55e21289))
* "Remember Last Selected Value" not working (frappe#21364) (#21395) ([9874c4a](https://github.com/ParsimonyGit/frappe/commit/9874c4a07baa9d354e96c2b8cc47eb8b195ffb18))
*  **UX:** Indicate that list view is empty because of filters (#21427) (#21428) ([2b24487](https://github.com/ParsimonyGit/frappe/commit/2b244875c0bd2a1cdcad2beef2365cff5dfa8f56))
* make sure number is not zero for `bankers_rounding` (#21431) (#21434) ([bbf8341](https://github.com/ParsimonyGit/frappe/commit/bbf8341b9ae4f4174d1030e0bea61122d1673350))
* sort options in group by field ([0ce6777](https://github.com/ParsimonyGit/frappe/commit/0ce6777d89bbde5a11fc6457cacad1c39bd56d1c))
* german translations ([3392a9c](https://github.com/ParsimonyGit/frappe/commit/3392a9c81c1f60630bc3fc6d492403a9356d42f8))
* make field label translatable ([a39dc90](https://github.com/ParsimonyGit/frappe/commit/a39dc90848346b38391bdb580212d27a2471ffa0))


### Features

* make option sorting configurable ([57f8076](https://github.com/ParsimonyGit/frappe/commit/57f8076988537ee9490c7183fbcf8fa5ef059540))
* enable "Sort Options" for Type field ([8b07c39](https://github.com/ParsimonyGit/frappe/commit/8b07c39911556d5a9c3474e059667b77cc848d7e))
* load more communications button in timeline ([d10fc9a](https://github.com/ParsimonyGit/frappe/commit/d10fc9adab4ee0c5a79d375823df2f0e2b55fd22))
* allow no_smtp_authentication to be set from site config (#21344) ([b498114](https://github.com/ParsimonyGit/frappe/commit/b49811462e1b85124880a80b2c97f1b27b22372e))


### Reverts

* temp fix to load communications ([9ad8b60](https://github.com/ParsimonyGit/frappe/commit/9ad8b60536ebd1611af7e5a37d47c51d1fc43427))


### Chore

* remove automatic transltion portal contribution ([7767ca0](https://github.com/ParsimonyGit/frappe/commit/7767ca06d68761a8798881fcaeda6c6ede6b89f7))
* resolve conflicts ([1debc26](https://github.com/ParsimonyGit/frappe/commit/1debc26ede54f6cc8dcfdbb6bf64dfba15ccd831))
* fix linter ([81cdeae](https://github.com/ParsimonyGit/frappe/commit/81cdeae22ca74a7e726c9c9c380969607df42a8d))
*  **release:** Bumped to Version 14.40.0 ([1a47007](https://github.com/ParsimonyGit/frappe/commit/1a47007d45f4c12d2c55c44e94fd24070bf3942d))
* add translation to label on new doc in Update quick_entry.js (#21444) (#21449) ([68ec5a6](https://github.com/ParsimonyGit/frappe/commit/68ec5a6883e04747854bfff625caf3370fd3a295))


### Performance Improvements

* index for_user in Notification log (#21379) (#21380) ([827b829](https://github.com/ParsimonyGit/frappe/commit/827b829aef251e3f4377d38a34538f16072e9fd3))
* faster docshare queries (backport #21392) (#21402) ([9af99ef](https://github.com/ParsimonyGit/frappe/commit/9af99efb95319977b3489392d4db5aef92bd466d))
* cache document naming rule to avoid recurring db calls (backport #18491) (#21408) ([03417ef](https://github.com/ParsimonyGit/frappe/commit/03417ef0eb3fc1de6d01cdbe706ea592cffa00b0))
* Faster report exporting logic (#21415) ([5c2ac47](https://github.com/ParsimonyGit/frappe/commit/5c2ac47142e6c30ccc2a76d01ac4a8b20d2f7703))


### Contributors

<a href="https://github.com/barredterra">barredterra</a>, <a href="https://github.com/shariquerik">shariquerik</a>, <a href="https://github.com/ankush">ankush</a>, <a href="https://github.com/mergify">mergify[bot]</a>, <a href="https://github.com/surajshetty3416">surajshetty3416</a>, <a href="https://github.com/elifvish">elifvish</a>, <a href="https://github.com/phot0n">phot0n</a>, <a href="https://github.com/frappe-bot">frappe-bot</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

ankush committed:
perf: faster doc shared checks

- If document, explicitly query document
- If checking doctype then put limit and only see if 1 record is
  returned.

(cherry picked from commit e4bae5c8315c25da3402739dfa2c615ba68667f1)

ankush committed:
test: add perf test helper for counting rows read

(cherry picked from commit c73d9fb7838178bb3d01db61ef3c76c38157b0dd)

barredterra committed:
fix: styling of group by button

(cherry picked from commit 85145c2c1189a538280c9cc4d4cf78acece2e629)

surajshetty3416 committed:
Merge pull request #21521 from frappe/mergify/bp/version-14-hotfix/pr-21519

mergify[bot] committed:
fix: correct condition check for dynamic filters (#21532)

Co-authored-by: Sagar Vora <sagar@resilient.tech>
Co-authored-by: Smit Vora <smitvora203@gmail.com>

surajshetty3416 committed:
Merge pull request #21535 from frappe/mergify/bp/version-14-hotfix/pr-21520

fix: styling of group by button (backport #21520)

ankush committed:
fix: duplicate hook for email pull (#21539)

surajshetty3416 committed:
Merge pull request #21462 from anhlevfx/version-14-hotfix

shariquerik committed:
chore: resolved conflicts

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-21425

mergify[bot] committed:
fix: ensure correct file path when importing package (#21569)

(cherry picked from commit d9167985b13a8d178d14ea46f16653218968774f)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-21425

shariquerik committed:
Merge pull request #21522 from frappe/mergify/bp/version-14-hotfix/pr-21425

fix: log errors while getting headers and data (backport #21425)

shariquerik committed:
fix: avoid this line if there is no tab break field in the form

(cherry picked from commit 8bf4008c5cead34ca02a539078e86f4c7efc8181)

barredterra committed:
chore: make isort ignore posthog (#21572)

shariquerik committed:
fix: load 21 communications to verify load more button is required or not.

(cherry picked from commit 76cc712f35a07bf15da4bdf137db2038b3cc1369)

shariquerik committed:
fix: render 20 communications + load more button if response has 21 communications

communication also has automated messages which was missed in last PR

(cherry picked from commit 61372f476d3b94cdc5b1782eaaf8d1c8cde5ecd6)

shariquerik committed:
fix: render load more button

(cherry picked from commit b09107b3a48981da90239de9f49cd0ee5ddb2b9b)

shariquerik committed:
Merge pull request #21590 from frappe/mergify/bp/version-14-hotfix/pr-21403

fix: missing filter on first connected nagivation (backport #21403)

shariquerik committed:
fix: pass webhook.request_url instead of undefined request_url

shariquerik committed:
Merge pull request #21591 from shariquerik/webhook-fix

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-21582

ankush committed:
Merge branch 'version-14' into version-14-hotfix

mergify[bot] committed:
perf: significant memory usage reductions (#21541)

* perf: reduce memory usage of background processes (#21467)

* perf: defer translation.py imports

This indirectly imports babel which isn't really required most of the
time.

* perf: defer gzip import

* perf: move validate_and_sanitize_search_inputs

This causes all sorts of indirect imports and increases memory usage

* perf: defer requests module imports

* perf: defer system settings import

* perf: defer LOG_DOCTYPES import

Causes many indirect imports

* perf: defer update_site_config

* perf: defer notifications import

* perf: remove unused import

* perf: defer safe exec import

* test: memory usage overhead

(cherry picked from commit 8a37d6d2786a35c4b6231aa029ce338790f1ecfe)

# Conflicts:
#	frappe/cache_manager.py
#	frappe/query_builder/utils.py
#	frappe/utils/__init__.py
#	frappe/utils/data.py

* chore: conflicts

* test: remove memory usage test

not measuring old stable branches

* refactor: Simplify FrappeOAuth2Client

No need to override anything, use request.session in way it is intended.

* perf: defer GoogleOAuth import

* perf: defer LDAP import

* perf: defer psutil import

* perf: defer pydoc import

* perf: defer QR code import

* perf: Freeze GC before forking Gunicorn workers

* fix: Only import LDAP if it's enabled

This is disabled on 99%+ sites, still incurs 3-4MB of memory hit on import of
ldap3.

* fix: move gc.freeze behind environ variable

* perf: Bump alloc count to 7,000 for generation 0

This has overall 1-2% CPU usage reduction for little to no costs.
Benefits increase when doing bulk processing with lots of objects.

* perf: Freeze GC right before starting background worker

BG worker forks are not CoW friendly. Freezing right before we start
worker should lessen overall memory usage. Though this isn't useful much
because at max you're sharing with 2 processes - master and horse.

WorkerPool can improve this benefit a lot by forking each worker from
master process and horse from forked processes. TBD when WorkerPool is
out of beta.

* fix: collect before freezing

* perf: defer loading JWT

* perf: Preload and share common python modules

* perf: preload more modules (#21557)

* perf: preload more modules

- bleach is used frequently for sanitization
- File gets imported anytime a private file is viewed. Indirect import
  of PIL is costly in each worker.

* test: warm up perf test

* test: fix broken tests

* fix: add missing import

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

Saboti215 committed:
feat: Add ability to use HTML fields in grids

(cherry picked from commit bcf837cbb5129d8858792a247b06d8eb4707fc6f)

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-21582

shariquerik committed:
Merge pull request #21601 from frappe/mergify/bp/version-14-hotfix/pr-21306

fix: always get parsed date value from a user date formatted date value (backport #21306)

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-21582

shariquerik committed:
Merge pull request #21587 from frappe/mergify/bp/version-14-hotfix/pr-21582

refactor: Load all communications (Email + Automated Messages) (backport #21582)

ankush committed:
Merge pull request #21588 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.40.2

## [14.40.2](https://github.com/frappe/frappe/compare/v14.40.1...v14.40.2) (2023-07-09)

### Bug Fixes

* always get parsed date value from a user date formatted date value ([5561e51](https://github.com/frappe/frappe/commit/5561e51c7050f4e5ec35115a228a2fc92b7cacbf))
* correct condition check for dynamic filters ([#21532](https://github.com/frappe/frappe/issues/21532)) ([0b2e055](https://github.com/frappe/frappe/commit/0b2e0552124be50a684788e02860e627b0401b46))
* duplicate hook for email pull ([#21539](https://github.com/frappe/frappe/issues/21539)) ([6ce37d5](https://github.com/frappe/frappe/commit/6ce37d52d7a38896805cd21d31608cbec73b2b14))
* ensure correct file path when importing package ([#21569](https://github.com/frappe/frappe/issues/21569)) ([1b52452](https://github.com/frappe/frappe/commit/1b5245252f08c79d7c97d8724e38ebdeb815a96a))
* load 21 communications to verify load more button is required or not. ([72bafb9](https://github.com/frappe/frappe/commit/72bafb97896612350b750a2583732ca27c5b92a4))
* log errors while getting headers and data ([7808fa1](https://github.com/frappe/frappe/commit/7808fa1a28c76fed456bbe636e72f113d43a24fb))
* **minor:** workflow state indicator ([#21508](https://github.com/frappe/frappe/issues/21508)) ([#21514](https://github.com/frappe/frappe/issues/21514)) ([978f501](https://github.com/frappe/frappe/commit/978f50122c05f93f9677315eb06e03a16ef56cf1))
* missing filter on first connected nagivation ([cdcdd5b](https://github.com/frappe/frappe/commit/cdcdd5b53f8ce7573b3494f3943c7a089f9f2562))
* pass webhook.request_url instead of undefined request_url ([53b751f](https://github.com/frappe/frappe/commit/53b751f2d08299b40e32586279ce5a9df766288e))
* removed redundant condition ([4c5d8ee](https://github.com/frappe/frappe/commit/4c5d8ee5340d29581c5f8d11efc3d7b5c89f0e95))
* render 20 communications + load more button if response has 21 communications ([a571172](https://github.com/frappe/frappe/commit/a5711728b4f5d84944a89ff0d3bf717f32afce8e))
* render load more button ([091c474](https://github.com/frappe/frappe/commit/091c474970114d193e214b091b09cc78035af072))
* repeat title value in description of link field. ([5957cbc](https://github.com/frappe/frappe/commit/5957cbcf0908261339e64bf6536e5c18af511ea1))
* return if exception occur before executing webhook ([b75a4af](https://github.com/frappe/frappe/commit/b75a4af739ffe5989da8fcb0820b90163ccb3208))
* styling of group by button ([de2f160](https://github.com/frappe/frappe/commit/de2f160a891669844117d988dade1a0d13fea27e))
* validate webhook secret ([82bfd5d](https://github.com/frappe/frappe/commit/82bfd5dab394ee7eb34e315a2c74d75508f3f891))

### Performance Improvements

* faster doc shared checks ([d3cb2c2](https://github.com/frappe/frappe/commit/d3cb2c21382b1c82b9557cd35f7a240f3690c50d))
* significant memory usage reductions ([#21541](https://github.com/frappe/frappe/issues/21541)) ([80d3d5c](https://github.com/frappe/frappe/commit/80d3d5cd69bbb9c12b04ea2c49aac9e07f1ebbcc)), closes [#21467](https://github.com/frappe/frappe/issues/21467) [#21557](https://github.com/frappe/frappe/issues/21557)

mergify[bot] committed:
fix: cast label to string before comparing (#21614) (#21620)



(cherry picked from commit e6c0b1cab5a45725331271f8e6b1694748da1d00)

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
chore: resolved conflict

shariquerik committed:
chore: resolved conflict

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-21419

shariquerik committed:
Merge pull request #21583 from frappe/mergify/bp/version-14-hotfix/pr-21419

fix: field replace (backport #21419)

sagarvora committed:
fix: ignore naming series validation when generating repeated address name (#21568)

(cherry picked from commit b4f1a2328b40a0c1f9288e33361ca5756e2d76df)

sagarvora committed:
Merge pull request #21623 from frappe/mergify/bp/version-14-hotfix/pr-21568

fix: ignore naming series validation when generating repeated address name (backport #21568)

Commit:
Merge remote-tracking branch 'frappe/version-14' into merge-v14

maharshivpatel committed:
fix: web form refresh_dependency on change

added refresh_dependency when value is changed in web form.

(cherry picked from commit 88450e6fdc65324a9a9a4d13d3372a27d2889062)

ankush committed:
Merge pull request #21632 from frappe/version-14-hotfix

chore: release v14

ankush committed:
Merge pull request #21631 from frappe/mergify/bp/version-14-hotfix/pr-21630

fix: web form refresh_dependency on change (backport #21630)

frappe-bot committed:
chore(release): Bumped to Version 14.40.3

## [14.40.3](https://github.com/frappe/frappe/compare/v14.40.2...v14.40.3) (2023-07-11)

### Bug Fixes

* avoid this line if there is no tab break field in the form ([aaf1002](https://github.com/frappe/frappe/commit/aaf1002fa705e2a3e508bc6a4db4aace92caacd7))
* cast label to string before comparing ([#21614](https://github.com/frappe/frappe/issues/21614)) ([#21620](https://github.com/frappe/frappe/issues/21620)) ([4c6d881](https://github.com/frappe/frappe/commit/4c6d881fe99ecf72b2a499ffb19d57106ff7d223))
* ignore naming series validation when generating repeated address name ([#21568](https://github.com/frappe/frappe/issues/21568)) ([c2bf114](https://github.com/frappe/frappe/commit/c2bf1149e099a62efaaa3acbf4e3e4bb7f3f56a4))
* replace field replacing correct dom element ([31bb02f](https://github.com/frappe/frappe/commit/31bb02f5927dfd974c067c7ca11db3998e8d7579))
* replace_field ([06ebaf8](https://github.com/frappe/frappe/commit/06ebaf82ed31f676bbda49937eda26e2645dae2e))

ankush committed:
Merge pull request #21637 from frappe/mergify/bp/version-14-hotfix/pr-21596

feat: Add ability to use HTML fields in grids (backport #21596)

shariquerik committed:
fix: translate autocomplete options label

(cherry picked from commit 94583340ec37e57e4354612d2b7d57555637783a)

# Conflicts:
#	frappe/public/js/frappe/form/controls/autocomplete.js

shariquerik committed:
fix: translate exported filename

(cherry picked from commit 8577bbbb3c89fc11702083534f2ab13aa3dfad77)

# Conflicts:
#	frappe/desk/utils.py

shariquerik committed:
chore: linter fix

(cherry picked from commit 3e012f4186a0ce1237932bf6e37fb5c934ebf73c)

mergify[bot] committed:
fix(setup_wizard): Don't send headers when exception happens on command line (#21641) (#21645)

Co-authored-by: Corentin Forler <8860073-cforler_dokos@users.noreply.gitlab.com>
(cherry picked from commit e8a57c63ce38c99d020230b2af5fd59648b32fd7)

Co-authored-by: Corentin Flr <10946971+cogk@users.noreply.github.com>

shariquerik committed:
Merge pull request #21644 from frappe/mergify/bp/version-14-hotfix/pr-21488

fix: calendar buttons on right get text translated (backport #21488)

shariquerik committed:
fix(WebForm): auto-increment link field (backport) + translate

shariquerik committed:
fix: translate filename before exporting

shariquerik committed:
Merge pull request #21650 from frappe/mergify/bp/version-14-hotfix/pr-21639

fix: translate autocomplete options label (backport #21639)

mergify[bot] committed:
ci: fix release notes generator (#21651) (#21653)

Without quotes it gets splatted and treated as separate args. Yes, bash.

(cherry picked from commit 17aad073d7b510756805ce4aeb9dee5eda0dfc7e)

# Conflicts:
#	.github/workflows/release_notes.yml

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix(Query Report): no double translation (#21606) (#21648)

(cherry picked from commit 218f7b473b65b4098cd0b0deecf6207a163efdc9)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
fix: session expiry logic (backport #21513) (#21615)

* fix: correct last update value

`NOW()` evalautes to server's time we should use system time instead.

(cherry picked from commit 564b960678d7ff31da1cb527f689a39a935a4016)

* refactor: use QB

(cherry picked from commit 7c4009fde98244459be25e1f6d15699f87f5f02c)

* fix: incorrect session expiry datediff

Datediff doesn't work like this in MYSQL, mysql just treats the
timestamp as flat timestamp.

(cherry picked from commit 60efb7c2ff7350f52a9da1f878ba14ae76d2230e)

# Conflicts:
#	frappe/sessions.py

* fix: Session insert using system time

NOW() is server time not system time.

(cherry picked from commit d353662b53aea7e4232de456bbcdf2de83f0fc99)

# Conflicts:
#	frappe/sessions.py

* refactor: Simplify expiry queries.

Dont rely on mysql dateutils, simply compare dates with a cutoff.

(cherry picked from commit 0e1236b6bef8109c561e524d73b51c3be92c9048)

# Conflicts:
#	frappe/sessions.py

* chore: conflicts

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
chore: remove unseen error message alert

Unnecessary and not useful for most users.

We should instead have dashboard for system status.

shariquerik committed:
fix: show workflow state in indicator if dont override status is set in workflow

(cherry picked from commit 2681364bca30d31d9b3607518c86d297e5ecd03c)

mergify[bot] committed:
perf(DX): Reduce recorder overhead (#21656) (#21658)

By processing `EXPLAIN` of queries in background after recording is
completed, the overhead reduces significantly and hence can be used
"more" in production environment too.

(cherry picked from commit 154a6dd9ed294100a29365e14f9d77e5291fa48f)

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
Merge pull request #21663 from frappe/mergify/bp/version-14-hotfix/pr-21657

fix: show workflow state in indicator if dont override status is set in workflow (backport #21657)

mergify[bot] committed:
fix: add numbers correctly in report column total (#21661) (#21662)

[skip ci]

(cherry picked from commit c2d35b773b1588fd95ca88e4a905c7df6b49ef23)

Co-authored-by: Nautatava Navlakha <nnautatva@gmail.com>

shariquerik committed:
revert: do not create comm on auto repeat #20606

(cherry picked from commit 75b26cc96bc08694f0ab038e52040b6ee925cad2)

ankush committed:
Merge pull request #21666 from frappe/mergify/bp/version-14-hotfix/pr-21665

revert: do not create comm on auto repeat #20606 (backport #21665)

mergify[bot] committed:
fix: `connect_replica` and `read_only` should be idempotent (#21674) (#21675)

* fix: `connect_replica` should be idempotent

Calling `connect_replica` twice ends up forgetting orginal writable
replica completely.

* test: replicas

closes https://github.com/frappe/frappe/issues/21619

(cherry picked from commit fefd9ac2e2190d37d3669390a2d6285506a2646c)

Co-authored-by: Ankush Menat <ankush@frappe.io>

Commit:
Merge remote-tracking branch 'frappe/version-14' into merge-v14

revant committed:
fix: setup_likes_from_feedback patch (#21677)

* fix: setup_likes_from_feedback patch

execute patch only if doctype feedback exists

* fix: check if table exists

---------

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>

ankush committed:
perf: lazy load sidebar stats

Sidebar stats take a huge time to load on large tables and it's not
required everytime you load the page. Best to only trigger it when user
clicks on "tags" (or other stats filter)

This is exactly how assignment and other filters work, so no need for
sidebar stats to work differently.

(cherry picked from commit d171a6d30aafdce594e6971ae3df1c5f12aca684)

ankush committed:
refactor: rename poorly named variables

stats are actually columns in DB
columns are for verifying if DB column exists hence db_columns
other stats was result.

Renamed everything to avoid confusion.

(cherry picked from commit 5841f296107680cfc1877d912ea90fa5902ebdf4)

ankush committed:
fix(UX): Show loading state while loading tags

(cherry picked from commit 9a9705bd1132dd0e37f87f5e68d0d5681f04af97)

ankush committed:
fix: Correct event for sidebar dropdown

- Click only gets called first time and not reliable next times
- Click will also get called when it's used for collapsing dropdown,
  which doesn't make sense here.

(cherry picked from commit b2b0bb69bb442f890547173e8727362db2a149d8)

ankush committed:
Merge pull request #21685 from frappe/mergify/bp/version-14-hotfix/pr-21683

perf: lazy load sidebar tag stats (backport #21683)

mergify[bot] committed:
feat: Custom naming series parser via hooks (#21690)

* feat: Custom naming series parser via hooks

* chore: use assignment operatot

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>

* test: Unit test for custom parser

* test: Unit test for custom parser

---------

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>
(cherry picked from commit e32e74f2f19dfc107a93da1acd30bf81a918ec17)

Co-authored-by: Deepesh Garg <deepeshgarg6@gmail.com>

shariquerik committed:
fix(UX): OAuth confirm access screen

(cherry picked from commit 539607d87a30106893f3e95d9704d571280471c4)

sagarvora committed:
fix: use current redirect URIs instead of legacy ones

(cherry picked from commit 4684e721428481d44ceae2e6e44686f13ed4392e)

# Conflicts:
#	frappe/integrations/doctype/oauth_client/oauth_client.json

sagarvora committed:
fix: conflict

sagarvora committed:
Merge pull request #21708 from frappe/mergify/bp/version-14-hotfix/pr-21707

fix: use current redirect URIs instead of legacy ones (backport #21707)

mergify[bot] committed:
test: Patch hooks for testing (#21702) (#21705)

* test: Patch hooks for testing

* test: hooks patcher

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit e38078a69f07e71422697b8730735940c0677dc0)

Co-authored-by: Deepesh Garg <deepeshgarg6@gmail.com>

shariquerik committed:
fix: do not show Home primary button

(cherry picked from commit cbe9ec26bd4e13f2d89f71ca246d17ff9b55bf85)

mergify[bot] committed:
fix: fetch portal sidebar items (#21703) (#21712)

(cherry picked from commit 149467a71b02ddfd42f3b03d7377bd802ab03837)

Co-authored-by: Gursheen Kaur Anand <40693548+GursheenK@users.noreply.github.com>

mergify[bot] committed:
fix: create version log for amended documents (backport #21680) (#21713)

* fix: compare with `doc.amended_from` while saving version for newly amended doc

(cherry picked from commit 7d7c440a1bd30a809243c1f523c0836b5e9298a2)

* fix: store version for name

(cherry picked from commit 86f0c4e52cd9614d5ccca5aa37d377555dc8bf8a)

---------

Co-authored-by: Sagar Vora <sagar@resilient.tech>

shariquerik committed:
fix: show list of user details that we are giving access to if scope is openid

(cherry picked from commit 264372a2a3344d7ffe8482b88fb6407bb3335393)

mergify[bot] committed:
build(deps): update pyyaml to 6.0.1 (#21728) (#21729)

(cherry picked from commit 03a582d7cdb81fe99b9a211136ded6cd8a703c27)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
Merge pull request #21716 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.41.0

# [14.41.0](https://github.com/frappe/frappe/compare/v14.40.3...v14.41.0) (2023-07-19)

### Bug Fixes

* `connect_replica` and `read_only` should be idempotent ([#21674](https://github.com/frappe/frappe/issues/21674)) ([#21675](https://github.com/frappe/frappe/issues/21675)) ([3151454](https://github.com/frappe/frappe/commit/31514545d33148a740ce5fc30dd9fe1d0062a35a))
* add numbers correctly in report column total ([#21661](https://github.com/frappe/frappe/issues/21661)) ([#21662](https://github.com/frappe/frappe/issues/21662)) ([e019190](https://github.com/frappe/frappe/commit/e0191906a8282d8e38eca4c16e6ff33e18530aae))
* calendar buttons on right get text translated ([82a08d9](https://github.com/frappe/frappe/commit/82a08d99716926b9258c7e66e80381ce08e23791))
* conflict ([25c8885](https://github.com/frappe/frappe/commit/25c8885e1e7db95bec3cc1d3f4628ae3cbafb3c1))
* Correct event for sidebar dropdown ([f9a254d](https://github.com/frappe/frappe/commit/f9a254d473341182551bb0d22dabf945b504d89e))
* create version log for amended documents (backport [#21680](https://github.com/frappe/frappe/issues/21680)) ([#21713](https://github.com/frappe/frappe/issues/21713)) ([1caff2c](https://github.com/frappe/frappe/commit/1caff2cf1be908cb8f2383ccf3fa3246a8e88bd8))
* fetch portal sidebar items ([#21703](https://github.com/frappe/frappe/issues/21703)) ([#21712](https://github.com/frappe/frappe/issues/21712)) ([1ab3782](https://github.com/frappe/frappe/commit/1ab3782ccb4a63aec5ec784a95eff0d10c1d061c))
* **Query Report:** no double translation ([#21606](https://github.com/frappe/frappe/issues/21606)) ([#21648](https://github.com/frappe/frappe/issues/21648)) ([91d2bd2](https://github.com/frappe/frappe/commit/91d2bd2065d31c6ff4bda589ddb03f3ee9d3e400))
* session expiry logic (backport [#21513](https://github.com/frappe/frappe/issues/21513)) ([#21615](https://github.com/frappe/frappe/issues/21615)) ([aad8e14](https://github.com/frappe/frappe/commit/aad8e147f9a01d30fa835933c0669a734071746b))
* setup_likes_from_feedback patch ([#21677](https://github.com/frappe/frappe/issues/21677)) ([39efdc7](https://github.com/frappe/frappe/commit/39efdc72e9c0f1522b41c799807431f2024b8381))
* **setup_wizard:** Don't send headers when exception happens on command line ([#21641](https://github.com/frappe/frappe/issues/21641)) ([#21645](https://github.com/frappe/frappe/issues/21645)) ([9f547b2](https://github.com/frappe/frappe/commit/9f547b23ae7d68e5d82c74c24c75ad03b7c93271))
* show workflow state in indicator if dont override status is set in workflow ([a985666](https://github.com/frappe/frappe/commit/a98566651a4b71a99fc9c8b0b5cf4646560d143c))
* translate autocomplete options label ([c36c836](https://github.com/frappe/frappe/commit/c36c836da4dd32a234725acfe809cb2ddf19f78a))
* translate exported filename ([f754b99](https://github.com/frappe/frappe/commit/f754b99c3367ce94bce32554cdb0f289b2eb71fa))
* translate filename before exporting ([f04062b](https://github.com/frappe/frappe/commit/f04062bae2822b3df156b21a12e8ef65c70d6a1e))
* use current redirect URIs instead of legacy ones ([2d2e57f](https://github.com/frappe/frappe/commit/2d2e57f433da0be92814e72b8421736948650fad))
* **UX:** Show loading state while loading tags ([8b2f86c](https://github.com/frappe/frappe/commit/8b2f86cc1dbfc7dbaa8fd9b3dc1f10dfc18df188))
* web form refresh_dependency on change ([e427bc0](https://github.com/frappe/frappe/commit/e427bc0a72a4a9814e80db4ac8e12dced46c7927))
* **WebForm:** auto-increment link field (backport) + translate ([719834c](https://github.com/frappe/frappe/commit/719834c8e2d383bc35eeb8afe07d7cf5c6259b49))

### Features

* Add ability to use HTML fields in grids ([2cebd1c](https://github.com/frappe/frappe/commit/2cebd1c754d729067bcea98407b135bf05fad46a))
* Custom naming series parser via hooks ([#21690](https://github.com/frappe/frappe/issues/21690)) ([4db6846](https://github.com/frappe/frappe/commit/4db68462c8d1b80dc0d9ffbcc8e11677fbf3a724))

### Performance Improvements

* **DX:** Reduce recorder overhead ([#21656](https://github.com/frappe/frappe/issues/21656)) ([#21658](https://github.com/frappe/frappe/issues/21658)) ([719d0bb](https://github.com/frappe/frappe/commit/719d0bb95dc3b3ae32c6d2b19fd200702786a3a3))
* lazy load sidebar stats ([9a88521](https://github.com/frappe/frappe/commit/9a88521c591903e5fc433f311dc19dc6b14e0fd4))

### Reverts

* do not create comm on auto repeat [#20606](https://github.com/frappe/frappe/issues/20606) ([dbdd786](https://github.com/frappe/frappe/commit/dbdd786efe969c0d7bba86d7cfa6e2f34d93d478))

mergify[bot] committed:
feat(DX): normalize queries in recorder (#21735) (#21736)

Attempt to normalize query by removing "variables"
This gives a different view of similar duplicate queries.

These two are distinct queries:
```sql
select * from user where name = 'x'
select * from user where name = 'z'
```

But their "normalized" form would be same:
```sql
select * from user where name = ?
```

This helps highlight queries ran in loop which might not register as
duplicate but are possibly "duplicate".

(cherry picked from commit 42aff950cebcd6c41038244ebddb4f3d76112f4c)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
perf: accurate cache eviction for assets (backport #21693) (#21739)

* fix: incorrect cache clearing of assets

- Build version wasn't correctly computed since v14 update of build
  system. This makes client side cache useless.
- We clear cache assuming rapid reloads,but opening new tab also does
  that. This makes the cache effectively useless for most users.

(cherry picked from commit 79532ea0f21d7502ba487c57e90e8a3de8225ce0)

* fix: Accurately determine rapid reloads

Assuming rapid-reload ~= user frustrated with old assets. We weren't
actually checking if load was "reload". This cleared cache when tabs
were opened and effectively made cache useless.

(cherry picked from commit 52c2e6d47bcd22d9a0a9dd2e4eb1cb6790478f96)

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix(UX): Highlight prepared reports older than 1 day (#21689) (#21741)

In MOST cases prepared report that are older than 1 day are stale.

[skip ci]

(cherry picked from commit 702e38cd74f16033a409066a1279a8bce6115390)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
perf(desk): reduce unnecessary requests on first load (backport #21695) (#21740)

* perf: eliminate request for user permissions

This is ALWAYS required, then it can just be part of boot. Eliminates 1
unnecessary request.

(cherry picked from commit b91961803216037aa60bbca71db522a5d4cbab09)

* perf: lazy load tag list for awesome bar

In last 7 days not a single person has used this feature on FC (based on
endpoint logs).

fix: When user knowingly triggers by typing text starting with `#` we
fire request to get tags lazily.

(cherry picked from commit 54b36e8604eed2538d12afddf9dabcc78ce4a54e)

* fix: lower rapid-reload threshold

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: actually debounce list view updates (#21738) (#21743)

Immediately debouncing and executing creates a new function everytime and doesn't actually "debounce" it just "delays". :moyai:

[skip ci]

(cherry picked from commit c2a5adadbe4176a18d539efffec75104d5601b53)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: don't allow setting 0 as max backups (#21747) (#21748)

This causes new backups to get deleted immediately.

If you want only 0 backups, then don't take backups.

(cherry picked from commit 96e6291692f0f6c3c6e18c8f7fef9d6565ae3409)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: Don't allow export of Virtual DocFields (#21749) (#21750)

(cherry picked from commit b2d32a2d639bb081e0a5f9c2e7e4a4449232ee86)

Co-authored-by: Gughan Ravikumar <gughanrk@gmail.com>

shariquerik committed:
Merge pull request #21746 from frappe/mergify/bp/version-14-hotfix/pr-21706

fix(UX): OAuth confirm access screen (backport #21706)

mergify[bot] committed:
perf: workflow transitions and bulk workflow (backport #21737) (#21751)

* fix: disable workflow emails by default

make it opt-in instead.

(cherry picked from commit d68e46d2cc5da3017a5b3924b9621380160198a6)

* refactor: Simpler workflow caching

Entire document is cached, so no need to create another layer of cache.

(cherry picked from commit 892c5e30a2804770d0cacd7660afa463bf103b1f)

# Conflicts:
#	frappe/model/workflow.py
#	frappe/workflow/doctype/workflow/workflow.py

* fix: disable list view updates during bulk approval

Also defer document refreshes by up to 5 seconds.

Rarely anyone needs truly realtime list view updates. It's better batch
them over at least 5 or so seconds if there's a high volume of changes.

(cherry picked from commit 06a905f6008a9c67129ed03d13fde3ccbba97bee)

* fix: duplicate on_row_checked call

(cherry picked from commit 948b24ee04ab5e54e448aaa4e59134618435cfb8)

* perf: Lazily compute common workflow transitions

(cherry picked from commit c90374c286e29758c399f33be9d1576f88b1d06c)

* chore: conflicts

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

Commit:
Merge remote-tracking branch 'frappe/version-14' into merge-v14

sagarvora committed:
fix: no API call required to set cookie in browser

(cherry picked from commit d2becc7f1d604a9e2fce7ebf099ccc9cf6cdd420)

# Conflicts:
#	frappe/translate.py

sagarvora committed:
chore: remove unused variable

(cherry picked from commit 5f0707c85659d7925456e0943cf95caac4e9ae53)

sagarvora committed:
Merge pull request #21772 from frappe/mergify/bp/version-14-hotfix/pr-21771

sagarvora committed:
Merge pull request #21784 from frappe/mergify/bp/version-14-hotfix/pr-21461

sagarvora committed:
chore: fix conflicts

sagarvora committed:
Merge pull request #21765 from frappe/mergify/bp/version-14-hotfix/pr-21763

fix: no API call required to set cookie in browser (backport #21763)

mergify[bot] committed:
perf: lazy load dashboard links (#21752) (#21791)

(cherry picked from commit 9608c32db7dd90a485b20a60f86ea899003b65db)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
perf: tune gc by default (#21804)

This is running on several prod site without any noticable problems, so
making it default behaviour.

Opt out by setting env variable `FRAPPE_TUNE_GC=False`

ankush committed:
Merge pull request #21801 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.42.0

# [14.42.0](https://github.com/frappe/frappe/compare/v14.41.0...v14.42.0) (2023-07-25)

### Bug Fixes

* actually debounce list view updates ([#21738](https://github.com/frappe/frappe/issues/21738)) ([#21743](https://github.com/frappe/frappe/issues/21743)) ([73342e0](https://github.com/frappe/frappe/commit/73342e0bde2fb8056247b308f4d3a4e93f063d66))
* do not show Home primary button ([3de9e6b](https://github.com/frappe/frappe/commit/3de9e6b029b33a6b18b9835372a7c800d86bb2b4))
* Don't allow export of Virtual DocFields ([#21749](https://github.com/frappe/frappe/issues/21749)) ([#21750](https://github.com/frappe/frappe/issues/21750)) ([1fc8088](https://github.com/frappe/frappe/commit/1fc80888b1c0d51e7f3c030284c338372bd41009))
* don't allow setting 0 as max backups ([#21747](https://github.com/frappe/frappe/issues/21747)) ([#21748](https://github.com/frappe/frappe/issues/21748)) ([956d6d5](https://github.com/frappe/frappe/commit/956d6d523ba97ff8968e921bbed38c8d88c7f5e2))
* no API call required to set cookie in browser ([19068c2](https://github.com/frappe/frappe/commit/19068c2e56d8b3623e45dba9a4108a65e3733d72))
* show list of user details that we are giving access to if scope is openid ([364b197](https://github.com/frappe/frappe/commit/364b1975c065fbf84eeedec661764205b5e23759))
* **UX:** Highlight prepared reports older than 1 day ([#21689](https://github.com/frappe/frappe/issues/21689)) ([#21741](https://github.com/frappe/frappe/issues/21741)) ([b1dce6c](https://github.com/frappe/frappe/commit/b1dce6cabb6c9e58d151b06ffee536513ea52779))
* **UX:** OAuth confirm access screen ([851361b](https://github.com/frappe/frappe/commit/851361b269fe5a3c3455b9b392831c66268338ea))

### Features

* before/after hooks for any app install/uninstall ([801d4b8](https://github.com/frappe/frappe/commit/801d4b893d948bc1230cfff1227cf9d0f557b689))
* **DX:** normalize queries in recorder ([#21735](https://github.com/frappe/frappe/issues/21735)) ([#21736](https://github.com/frappe/frappe/issues/21736)) ([e52b2d9](https://github.com/frappe/frappe/commit/e52b2d946c8f1e9ba85854857ab7e215e2d764fc))

### Performance Improvements

* accurate cache eviction for assets (backport [#21693](https://github.com/frappe/frappe/issues/21693)) ([#21739](https://github.com/frappe/frappe/issues/21739)) ([f02ae6e](https://github.com/frappe/frappe/commit/f02ae6e5b38474b63f4ccdeff43a84e3ef0cab96))
* **desk:** reduce unnecessary requests on first load (backport [#21695](https://github.com/frappe/frappe/issues/21695)) ([#21740](https://github.com/frappe/frappe/issues/21740)) ([56d4166](https://github.com/frappe/frappe/commit/56d41661cba6ded7321deaf094562c70a7b0dd8f))
* lazy load dashboard links ([#21752](https://github.com/frappe/frappe/issues/21752)) ([#21791](https://github.com/frappe/frappe/issues/21791)) ([f8cc3bd](https://github.com/frappe/frappe/commit/f8cc3bd2a61b6fb424c8fd137313f3d740dbc4be))
* tune gc by default ([#21804](https://github.com/frappe/frappe/issues/21804)) ([0b1b54e](https://github.com/frappe/frappe/commit/0b1b54e7217a64a86cacab36f8c32242d65671c6))
* workflow transitions and bulk workflow (backport [#21737](https://github.com/frappe/frappe/issues/21737)) ([#21751](https://github.com/frappe/frappe/issues/21751)) ([f3e933d](https://github.com/frappe/frappe/commit/f3e933d497ae63c38c647a43b7d85cc15b9c7c7b))

Commit:
Merge remote-tracking branch 'frappe/version-14' into merge-v14
